### PR TITLE
Struct support for CSV and YAML exporters

### PR DESCRIPTION
**Pull request by kkoppolu1**
## Description
Export struct data types in YAML and CSV. The implementation is identical to JSON.

### Other changes:
1. Exclude generated CSV files from trailing whitespace check of precommit hook
2. Add "default" attribute of the VSS node to the CSV exporter
3. Remove duplicate newline at the end of CSV file added by the CSV exporter

## Testing
- Added and updated unit tests
## Issues
https://github.com/COVESA/vss-tools/issues/236
https://github.com/COVESA/vss-tools/issues/237

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ repos:
     rev: v3.2.0
     hooks:
     -   id: trailing-whitespace
+        exclude_types: ["csv"]
```
Comment (erikbosch) [inline .pre-commit-config.yaml:8]: I see it as an important point to agree on next meeting on how we want to handle line breaks and what files to exclude. In some of the other PRs I added a rule to exclude all files starting with "expected*" for the same purpose. I think both methods work. If we do not want to add a line break for JSON files then we shall likely exclude all JSON files as well.

**Comment by erikbosch** [comment]
Meeting notes:

- Please review, merge decision next week
- Sebastian: It would be good if we document how output would look like somewhere. 
- Erik: Possibly extend https://github.com/COVESA/vss-tools/blob/master/docs/vspec2x.md to include some information.
- Erik to check with Krishna if Erik or Krishna shall add documentation.

File: vspec2x.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ def processDataTypeTree(parser: argparse.ArgumentParser, args, include_dirs,
     if len(args.overlays) > 0:
         parser.error(
             "Overlays are not yet supported in vspec struct/data type support feature")
-    if args.format.name != "json":
+    if args.format.name not in ["json", "yaml", "csv"]:
```
[the remark this answers is not in the document]
Comment (kkoppolu1) [reply vspec2x.py]: Possible. The downside is that invalid command line arguments are not validated upfront before any processing is done. It is a trade-off between more code and different user experience. If you prefer lazy validation, I can change it.

[the remark this answers is not in the document]
Comment (erikbosch) [reply vspec2x.py]: I have no strong opinion. My main though was to only have a single location where we do the check and list which formatters that support types. An alternative approach could be to keep the first check, but there set data_type_tree to None if using an exporter that does not support types and just check for data_type_tree here, or keep it as is :-)

**Comment by kkoppolu1** [comment]
> Meeting notes:
> 
> * Please review, merge decision next week
> * Sebastian: It would be good if we document how output would look like somewhere.
> * Erik: Possibly extend https://github.com/COVESA/vss-tools/blob/master/docs/vspec2x.md to include some information.
> * Erik to check with Krishna if Erik or Krishna shall add documentation.

The test CSV/YAML files in this PR give an indication of the output file structure for the purposes of code review.
@erikbosch I can work on the documentation (I will probably create a separate PR). I will follow the suggestion to extend https://github.com/COVESA/vss-tools/blob/master/docs/vspec2x.md . Let me know.

**Comment by erikbosch** [comment]
@kkoppolu1 - sounds like a good approach to handle the documentation in a separate PR.

**Pull request merged**